### DISCUSSION TITLE
What does "<defaultAlias>" mean?

### DISCUSSION THREAD

**Discussion by leira** [ArturoDent/command-alias#3] [Q&A] [answered]
I don't understand `<defaultAlias>`. What does it mean to create an alias with `<defaultAlias>`? What alias should I type in the Command Palette to run this alias? If multiple alias with `<defaultAlias>` is defined, I can see only one entry in the Command Palette. If I select it, which command will run?

**Comment by ArturoDent** [comment, marked as answer]
Sorry, I will try to clarify the `<defaultAlias>` description.

If you make a new alias with the command: `*Create aliases from vscode's built-in commands` and **do not** actually provide an alias - like `touch` for the command `explorer.newFile` then an alias which is literally the string `<defaultAlias>` will be used as the alias.  You can see this in the following gif where I created an alias with just an empty string.


![defaultAlias](https://user-images.githubusercontent.com/31813701/147980456-4d1830ff-0a42-4e19-8c4a-f340379c6eac.gif)

Of course it makes no sense to try to create an alias with an empty string - and if you try to do it manually in the settings like

```jsonc
  "command aliases": {
    "explorer.newFolder": ""
  }
```

you will get a message about the problem.

The `<defaultAlias>` is really intended as a one-time temporary solution only.  I could make the command just fail and will look into adding a warning or error message.

If you attempt to make more than one command with an empty string alias, the extension itself will fail on reload with the error message that that command `<defaultAlias`> already exists.  Then only the first command created by you (and thus registered as a command by this extesnion) with that `id` of `<defaultAlias>` would appear in the Command Palette and run.
